src/transformStateWithClones.js

```diff
@@ -8,6 +8,51 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const statesHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = addProperties(currentState, action.extraData);
+        statesHistory.push(currentState);
+        break;
+      case 'removeProperties':
+        currentState = removeProperties(currentState, action.keysToRemove);
+        statesHistory.push(currentState);
+        break;
+      case 'clear':
+        currentState = clearProperties(currentState);
+        statesHistory.push(currentState);
+        break;
+    }
```

Comment:
[CODE STYLE]: Use a `default` case in your `switch` statement for error handling. It's important to have a fallback for unexpected action types.